prime-router/src/main/kotlin/history/db/ReportGraph.kt

```diff
@@ -172,19 +171,19 @@ class ReportGraph(
             .from(cte)
             .join(REPORT_FILE)
             .on(REPORT_FILE.REPORT_ID.eq(ItemGraphTable.ITEM_GRAPH.PARENT_REPORT_ID))
-            .join(ACTION)
-            .on(ACTION.ACTION_ID.eq(REPORT_FILE.ACTION_ID))
-            .where(ACTION.ACTION_NAME.eq(TaskAction.receive))
+            .leftJoin(REPORT_LINEAGE)
+            .on(REPORT_FILE.REPORT_ID.eq(REPORT_LINEAGE.CHILD_REPORT_ID))
+            .where(REPORT_LINEAGE.PARENT_REPORT_ID.isNull())
+            .orderBy(REPORT_FILE.ACTION_ID.asc())
```

Comment:
The previous query was to join the `action` table in order to check if the `action_name` is `receive`. The new query instead filters the list of reports to only include those that do not appear in `report_lineage` as a child. This list is then ordered by `action_id` to ensure the results are output in a consistent order. The join to the `action` table is removed since it is no longer necessary.

Reply:
Thanks, this explanation was very helpful!

Reply:
Just to clarify, we are now recursively going up the report lineage table until we find a parent report id which is null?

Reply:
The recursive query (`cte`) was not altered, but that is the current and previous behavior. The change here was the in the way the results from the recursive query are filtered.

Reply:
Got it 👍